source/templates.tex

```diff
@@ -5698,8 +5698,9 @@
 An instantiated template specialization can be either implicitly
 instantiated\iref{temp.inst} for a given argument list or be explicitly
 instantiated\iref{temp.explicit}.
-A specialization is a class, variable, function, or class member that is either
-instantiated or explicitly specialized\iref{temp.expl.spec}.
+A \defn{specialization} is a class, variable, function, or class member that is either
+instantiated\iref{temp.inst} from a template or produced by an 
+explicit specialization\iref{temp.expl.spec} of a template.
```

Comment:
I am having second thoughts about "produced by ...". In fact, an explicit specialization *is* a class etc. right there; it doesn't obviously "produce" one. (Oh, and "produce" sounds awkward anyway.)
maybe "... or an explicit specialization of a template."
@zygoloid ?

Reply:
Looks to me now.

Reply:
Hmm, are the "from a template" additions correct? For cases like:

```
template<typename T> struct X {
  void f() {}
};
```

... is `X<int>::f()` instantiated from a template, or from a member of a template?

Reply:
Also, "is either instantiated from a template or an explicit specialization of a template" is missing a verb. Or *what* an explicit specialization? (If this is "or is an explicit specialization", what does that mean for members of templates:

```
template<typename T> struct Y {
  struct Z {};
};
template<> struct Y<int>::Z {};
template<> struct Y<float> { struct Z {}; };
```

... which of `Y<int>::Z` and `Y<float>::Z` are specializations? Do we really mean that the former is a specialization and the latter is not?

Reply:
@zygoloid I think both of these could be addressed by saying:

> A *specialization* is a class, variable, function, or class member that is either instantiated from a template, or a member thereof, or is an explicit specialization of a template, or member thereof.

This seems a little clunky though.

Reply:
"templated entity" to the rescue, it seems. (This is actually a defined term.)

Reply:
Templated entity would work, good idea. Looks like the note http://eel.is/c++draft/temp#pre-8.note-1 is wrong though, since a variable is not an entity.

Reply:
A variable declares an object or reference, and both of these things are entities. [basic.pre].

Reply:
Correct, but the variable itself is not

Reply:
Well, it was certainly intended that variable templates are also templated entities. Maybe we should simply say that a variable is an entity, then.

Reply:
A variable template is a template though, so it is a templates entity anyways.

Reply:
So, we're good with "templated entity", and all we need is an updated pull request.

Reply:
Updated